### PR TITLE
[fix] Re-prompt for permissions if necessary (issue #678)

### PR DESCRIPTION
**Pull request by AdiAr11**
### Types
- [x] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (non-breaking change which improves code quality - QA thoroughly )
- [ ] Breaking Change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (updates that would not affect or change the code involving the app itself)

#### 🐛 Fixes
- Fix issue #678
### QA

- [ ] Signup
- [ ] Login
- [ ] Logout
- [ ] Unencrypted Chat 
- [ ] Encrypted Chat 
- [ ] Group Chats
- [ ] Profile Changes
- [ ] Theming Changes
- [ ] Force Kill and Restart

### Final Checklist
 
- [x] All associated issues have been linked to PR
- [ ] All necessary changes made to the documentation
- [ ] Parties interested in these changes have been notified
- [ ] Linter has been run on all code in the PR

### CODE REVIEW

File: lib/views/home/chat/widgets/chat-input.dart [hunk 1]
```diff
@@ -210,6 +211,19 @@ class ChatInputState extends State<ChatInput> {
...
+    const photosPermission = Permission.photos;
+    final status = await photosPermission.status;
+
+    if (!status.isGranted) {
+      openAppSettings();
+    }else{
+      onAddPhoto();
+    }
+
+  }
```
Comment (EdGeraghty) [inline lib/views/home/chat/widgets/chat-input.dart:225]: May I recommend you split this method into two?

`checkPhotoPermission` implies returning a boolean to me, the actual functionality of the method is not obvious from the name

File: lib/views/home/chat/widgets/chat-input.dart [hunk 2]
```diff
@@ -586,7 +600,7 @@ class ChatInputState extends State<ChatInput> {
                           child: MediaCard(
                             text: Strings.buttonGallery,
                             icon: Icons.photo,
-                            onPress: () => onAddPhoto(),
+                            onPress: () => checkPhotoPermission(),
```
Comment (EdGeraghty) [inline lib/views/home/chat/widgets/chat-input.dart:603]: I'd move the functionality in here

**Review by EdGeraghty — CHANGES_REQUESTED** [review]

File: lib/views/home/chat/widgets/chat-input.dart (hunk 2 above)
Comment (AdiAr11) [reply lib/views/home/chat/widgets/chat-input.dart:603]: Okay, I'll make these changes 😀

Comment (EdGeraghty) [reply lib/views/home/chat/widgets/chat-input.dart:603]: Perfect!

File: lib/views/home/chat/widgets/chat-input.dart (hunk 1 above)
Comment (EdGeraghty) [reply lib/views/home/chat/widgets/chat-input.dart:225]: Perhaps something like `hasPhotoPermission` may be even more explicit

File: lib/views/home/chat/widgets/chat-input.dart [hunk 3]
```diff
@@ -586,7 +587,15 @@ class ChatInputState extends State<ChatInput> {
                           child: MediaCard(
                             text: Strings.buttonGallery,
                             icon: Icons.photo,
-                            onPress: () => onAddPhoto(),
+                            onPress: () async{
+                              const photosPermission = Permission.photos;
+                              final status = await photosPermission.status;
+                              if(!status.isGranted){
+                                openAppSettings();
```
Comment (EdGeraghty) [inline lib/views/home/chat/widgets/chat-input.dart:594]: It's probably a good idea to warn the user that you're about to throw them to the app settings, and to tell them why.

Search for `showDialog` in the codebase to see how to add a warning - and if you're going to add text, that seems like a great time to add something to our internationalisation (i18n) library!

**Review by EdGeraghty — CHANGES_REQUESTED** [review]

**Comment by EdGeraghty** [comment]
Sorry for the faff - had a realisation after setting ready for review!

File: lib/views/home/chat/widgets/chat-input.dart (hunk 3 above)
Comment (EdGeraghty) [reply lib/views/home/chat/widgets/chat-input.dart:594]: Perfect!

File: lib/global/strings.dart [hunk 4]
```diff
@@ -397,6 +400,7 @@ class Strings {
   static final contentRenameDevice = tr('content-dialog-device-rename');
   static final contentKeyExportWarning =
       tr('content-dialog-devices-key-export');
+  static final contentPhotoPermission = tr('content-dialog-photo-permission');
```
Comment (EdGeraghty) [inline lib/global/strings.dart:403]: Good work on the i18n front!

File: lib/views/home/chat/widgets/chat-input.dart [hunk 5]
```diff
@@ -230,6 +231,56 @@ class ChatInputState extends State<ChatInput> {
     onToggleMediaOptions();
   }
 
+  showDialogForPhotoPermission(BuildContext context){
+    if(Platform.isAndroid) {
+      showDialog(
```
Comment (EdGeraghty) [inline lib/views/home/chat/widgets/chat-input.dart:236]: I'm not so sure we should differentiate here, we should probably just use showDialog() regardless of platform.

Speaking of which, have you tried this on a desktop build?

File: lib/views/home/chat/widgets/chat-input.dart [hunk 6]
```diff
@@ -591,7 +642,7 @@ class ChatInputState extends State<ChatInput> {
                               const photosPermission = Permission.photos;
                               final status = await photosPermission.status;
                               if(!status.isGranted){
-                                openAppSettings();
+                                showDialogForPhotoPermission(context);
```
Comment (EdGeraghty) [inline lib/views/home/chat/widgets/chat-input.dart:645]: Unambiguous. Nice.

File: lib/global/strings.dart (hunk 4 above)
Comment (AdiAr11) [reply lib/global/strings.dart:403]: Thank you !

File: lib/views/home/chat/widgets/chat-input.dart (hunk 5 above)
Comment (AdiAr11) [reply lib/views/home/chat/widgets/chat-input.dart:236]: Okay, I'll use the showDialog function only. 
Also I haven't tried the desktop build yet

**Review by EdGeraghty — APPROVED** [review]
👍

**Review by ereio — APPROVED** [review]
Awesome work, thanks so much for contributing! 👍 :)

**Comment by AdiAr11** [comment]
> Awesome work, thanks so much for contributing! 👍 :)

Thank you so much !!

**Pull request merged**